ptask: /build/ptask-1.0.0/src/tw.c:166: undefined reference to `is_error'

## Issue description

ptask seems to be broken on master rev 44a4370b1f3aa2b1b2416d8c09b8df6900f5c449

### Steps to reproduce

```
$ nix-shell -p ptask
[...]

ui_tasktree.c: In function 'tasktree_button_press_event_cbk':
ui_tasktree.c:434:3: warning: 'gtk_menu_popup' is deprecated: Use '(gtk_menu_popup_at_widget, gtk_menu_popup_at_pointer, gtk_menu_popup_at_rect)' instead [-Wdeprecated-declarations]
   gtk_menu_popup(w_menu,
   ^~~~~~~~~~~~~~
In file included from /nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/gtklabel.h:34:0,
                 from /nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/gtkaccellabel.h:35,
                 from /nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/gtk.h:33,
                 from ui_tasktree.c:25:
/nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/gtkmenu.h:138:9: note: declared here
 void    gtk_menu_popup    (GtkMenu        *menu,
         ^~~~~~~~~~~~~~
ui_tasktree.c: In function 'tasktree_visible_activate_cbk':
ui_tasktree.c:448:2: warning: 'gtk_action_get_name' is deprecated [-Wdeprecated-declarations]
  aname = gtk_action_get_name(action);
  ^~~~~
In file included from /nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/deprecated/gtkactivatable.h:25:0,
                 from /nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/gtk.h:248,
                 from ui_tasktree.c:25:
/nix/store/r6hq8arfhi98rdq3gf7zqswkciaildg8-gtk+3-3.22.30-dev/include/gtk-3.0/gtk/deprecated/gtkaction.h:103:14: note: declared here
 const gchar* gtk_action_get_name               (GtkAction     *action);
              ^~~~~~~~~~~~~~~~~~~
gcc  -g -O2 -Wl,--as-needed  -export-dynamic  -o ptask log.o main.o note.o pio.o pstr.o ptime.o settings.o tw.o ui.o ui_newtask_diag.o ui_projecttree.o ui_taskpanel.o ui_tasktree.o  -L/nix/store/sfvxpy6c7hd5jdn8di8dkd6rx95bfxrc-json-c-0.13.1/lib -ljson-c -L/nix/store/f4myj2j6j6lyjsgqymnnb8z7yh7zzw43-gtk+3-3.22.30/lib -L/nix/store/ycpkd4jxpy913v6fxf4b5mmy9bz81vsy-glib-2.56.0/lib -L/nix/store/03slwm4ni6hlrkpishs37njicyxh2ywg-cairo-1.15.12/lib -L/nix/store/wrxdkfnvkdvjbsf1gjs1ig2z5k8j3w5d-pango-1.42.1/lib -L/nix/store/vdkxi8rgkwzhdz6pwi5rp4js4mr1mk7k-fribidi-1.0.4/lib -L/nix/store/ld0404gxvs9rncrnrqywhvyymciavjlm-gdk-pixbuf-2.36.12/lib -L/nix/store/xprys13vkavfv64g3v72zgh0974dp9r6-atk-2.28.1/lib -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lfribidi -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0
tw.o: In function `task_exec_json':
/build/ptask-1.0.0/src/tw.c:166: undefined reference to `is_error'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:471: ptask] Error 1
make[3]: Leaving directory '/build/ptask-1.0.0/src'
make[2]: *** [Makefile:557: all-recursive] Error 1
make[2]: Leaving directory '/build/ptask-1.0.0/src'
make[1]: *** [Makefile:485: all-recursive] Error 1
make[1]: Leaving directory '/build/ptask-1.0.0'
make: *** [Makefile:384: all] Error 2
builder for '/nix/store/72csq7lpdwqdfvr6rp067d5ac8axkc4n-ptask-1.0.0.drv' failed with exit code 2
error: build of '/nix/store/72csq7lpdwqdfvr6rp067d5ac8axkc4n-ptask-1.0.0.drv' failed
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 4.17.13, NixOS, 18.09.git.2038ac39f5f (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos"`
 - channels(kalbasit): `"nixos-18.09pre145679.dae9cf6106d"`
 - nixpkgs: `/code/personal/base/src/github.com/kalbasit/system/external/nixpkgs`

--- a/pkgs/applications/misc/ptask/default.nix
+++ b/pkgs/applications/misc/ptask/default.nix
@@ -12,7 +12,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ pkgconfig makeWrapper ];
 
-  patches = [ ./tw-version.patch ];
+  patches = [ ./tw-version.patch ./json_c_is_error.patch ];
 
   preFixup = ''
     wrapProgram "$out/bin/ptask" \

--- a/pkgs/applications/misc/ptask/default.nix
+++ b/pkgs/applications/misc/ptask/default.nix
@@ -25,5 +25,6 @@ stdenv.mkDerivation rec {
     description = "GTK-based GUI for taskwarrior";
     license = licenses.gpl2;
     maintainers = [ maintainers.spacefrogg ];
+    platforms = platforms.linux;
   };
 }

--- a/pkgs/applications/misc/ptask/json_c_is_error.patch
+++ b/pkgs/applications/misc/ptask/json_c_is_error.patch
@@ -1,0 +1,13 @@
+diff --git a/src/tw.c b/src/tw.c
+index 602f7b3..ca601cd 100644
+--- a/src/tw.c
++++ b/src/tw.c
+@@ -163,7 +163,7 @@ static struct json_object *task_exec_json(const char *opts)
+ 
+ 	free(cmd);
+ 
+-	if (o && is_error(o))
++	if (!o)
+ 		return NULL;
+ 
+ 	return o;

